### PR TITLE
Consider volecity of the EndDragging. Act more like Apple style.

### PR DESCRIPTION
**Pull request by imzark**

### CODE REVIEW

**Comment by MaherKSantina** [comment]
Hello @imzark ! Thank you so much for giving interest to this repo! And thank you for raising this PR! 🎉  I'll add my comments inline

File: MSPeekCollectionViewDelegateImplementation/Classes/MSPeekCollectionViewDelegateImplementation.swift [hunk 1]
```diff
@@ -80,23 +80,32 @@ open class MSPeekCollectionViewDelegateImplementation: NSObject {
...
+            
+            if coefficent > 0 {
+                coefficent = max(minimumItemsToScroll, coefficent)
+            } else if coefficent < 0 {
+                coefficent = min(-minimumItemsToScroll, coefficent)
+            }
+            
+        } else {
+            
+            coefficent = Int(scrollDistance / ((scrollWidth + cellSpacing) / 2))
```
Comment (MaherKSantina) [inline MSPeekCollectionViewDelegateImplementation/Classes/MSPeekCollectionViewDelegateImplementation.swift]: It looks like the number of items to scroll is being ignored when the velocity is negative, is that the intended behavior?

Comment (imzark) [reply MSPeekCollectionViewDelegateImplementation/Classes/MSPeekCollectionViewDelegateImplementation.swift]: Thanks for  reply! @MaherKSantina 

And sorry for my uncautious code. The condition should be `abs(velocity) > 0`.

So my intention is when finger drag ending with no velocity (not negative velocity) , the `destinationIndex` may be judged by (scrollDistance / halfPageWidth). 

Now in the repo, the `destinationIndex` is always judged by (scrollDistance / scrollThreshold), that makes scrolling to next page too easy when the velocity is zero. That means if I drag a little distance (but more than `scrollThreshold`) and stop my finger, the collectionView will scroll to next page.

**Comment by imzark** [comment]
Thanks for reply! @MaherKSantina

And sorry for my uncautious code. The condition should be `abs(velocity) > 0`.

So my intention is when finger drag ending with no velocity (not negative velocity) , the `destinationIndex` may be judged by (scrollDistance / halfPageWidth).

Now in the repo, the `destinationIndex` is always judged by (scrollDistance / scrollThreshold), that makes scrolling to next page too easy when the velocity is zero. That means if I drag a little distance (but more than `scrollThreshold`) and stop my finger, the collectionView will scroll to next page.

**Comment by MaherKSantina** [comment]
Ohh haha thanks @imzark for the clarification! 
I actually just tried the app store paging and it looks like it also moves to the next page even if you slightly scroll with your finger. 
That being said, the scroll distance in the repo is somewhat taking the velocity into consideration because it's using the `targetContentOffset`. If the distance between the `targetContentOffset` and the current scroll is large, that means there is high velocity.

Would love your input on this!

**Comment by imzark** [comment]
Yeah,, I'm not sure whether you touch the little difference I was saying... cause me not that good at english expression.

gifs below explain my thought.

old

new

**Comment by MaherKSantina** [comment]
@imzark ohh I see what you mean, but I also tried this on the app store and here's a video of the behavior:
ezgif-2-2cd24c0c4a3d

It looks like it's switching to the other page even without velocity. Is that what you're talking about?

**Comment by imzark** [comment]
Yes! Without velocity, It not always scroll to next page.

**Comment by MaherKSantina** [comment]
So you're saying that you don't want it to scroll all the time to the next page if there is no velocity?
Wouldn't increasing the `scrollThreshold` fix this?

**Comment by imzark** [comment]
Thanks again for your reply! And ya you got my point~ 😁

Increasing `scrollThreshold` would make it more difficulty to scroll if there is a velocity.

So I mean may be another way is needed to decide whether scroll to next page when there is no velocity. That is something like `scrollDistance / halfPageWidth`, just like the App Store's behavior.

**Comment by MaherKSantina** [comment]
Ohhh I think now I understand fully what you're saying! Yeah that's a good idea! I was thinking maybe having a `velocityThreshold` similar to the scroll threshold and that would dictate whether we need to move to the next page or not based on the threshold. What do you think?

**Comment by imzark** [comment]
YEAH! That would be a better solution I think.

**Comment by MaherKSantina** [comment]
Awesome! Do you think you can update this PR to reflect what we talked about? 👼

**Comment by imzark** [comment]
Sure, I'll give a try

**Comment by imzark** [comment]
Hello @MaherKSantina ! I've update the PR, would you please test it? 😁

**Comment by MaherKSantina** [comment]
Awesome @imzark! 🎉  Thank you very much! I'll take a look at today if I have time after work!
Really appreciate the effort 🎖

**Comment by MaherKSantina** [comment]
Hello @imzark ! Apologies for the delay but I've been very busy at work 😢 I'll try to take a look at it as soon as I can! Thank you for your patience

**Comment by imzark** [comment]
Yeah it's ok, take your time. Just a little change btw

**Comment by MaherKSantina** [comment]
Hey @imzark ! I just wanted to let you know that I completely re-developed the library using a custom collection view layout and a separate paging class https://github.com/MaherKSantina/MSPeekCollectionViewDelegateImplementation/releases/tag/3.0.0. The reason is that people are having a lot of issues with the current implementation. So hopefully the new version is better!

I tried to incorporate your logic in this PR in the new version, but I'm not 100% sure I got it right. Do you mind taking a look at the new logic and letting me know if it's working the way you want it to?

In the meantime I'll close this PR because the logic and location of code has changed, but if you found out that there's something that can be enhanced please feel free to create a new PR

Much appreciated buddy ❤️

**Comment by imzark** [comment]
Awesome! 👏 I would be glad to try the new version and let you know if meeting any problem. Thanks for the excellent repo. 👍

**Comment by MaherKSantina** [comment]
@imzark yes please let me know and thank you so much for your support! 🎉

**Pull request closed without merging**